go/store/types/value_store.go

```diff
@@ -794,6 +761,7 @@ func (lvs *ValueStore) gc(ctx context.Context,
 		cErr := sweeper.Close(ctx)
 		return nil, errors.Join(err, cErr)
 	}
+	final = nil
```

Comment:
Why is it important to set this local var to nil?

Reply:
I think it's dumb on my part and in reality the compiler + garbage collector know / can know that the variable is no longer live after that. But, since it's potentially large and we are going on to an arbitrary amount of work where it does not need to be live, we nil it out...